File: server/src/routes/balancingRoutes.js
/**
 * Pseudocode:
 * 1) Define balancing toolkit endpoints.
 * 2) Protect them for designers.
 */

export const balancingRoutes = [
  {
    method: "GET",
    path: "/balancing/parameters",
    handlerName: "BalancingController.getParameters",
    requiresRole: "designer",
  },
  {
    method: "POST",
    path: "/balancing/parameters",
    handlerName: "BalancingController.saveParameters",
    requiresRole: "designer",
  },
  {
    method: "GET",
    path: "/balancing/suggestions",
    handlerName: "BalancingController.suggestions",
    requiresRole: "designer",
  },
  {
    method: "POST",
    path: "/balancing/simulate",
    handlerName: "BalancingController.simulate",
    requiresRole: "designer",
  },
  {
    method: "GET",
    path: "/decisions",
    handlerName: "BalancingController.decisions",
    requiresRole: "designer",
  },
  {
    method: "POST",
    path: "/decisions",
    handlerName: "BalancingController.createDecisionEntry",
    requiresRole: "designer",
  },
];

File: server/src/routes/BalancingRoutes.java
package routes;

import java.util.List;
import types.Types;

/**
 * Pseudocode:
 * 1) Define balancing toolkit endpoints.
 * 2) Protect them for designers.
 */
public class BalancingRoutes {
  public static final List<Types.RouteDefinition> ROUTES = List.of(
    new Types.RouteDefinition(
      Types.HttpMethod.GET,
      "/balancing/parameters",
      "BalancingController.getParameters",
      Types.Role.DESIGNER
    ),
    new Types.RouteDefinition(
      Types.HttpMethod.POST,
      "/balancing/parameters",
      "BalancingController.saveParameters",
      Types.Role.DESIGNER
    ),
    new Types.RouteDefinition(
      Types.HttpMethod.GET,
      "/balancing/suggestions",
      "BalancingController.suggestions",
      Types.Role.DESIGNER
    ),
    new Types.RouteDefinition(
      Types.HttpMethod.POST,
      "/balancing/simulate",
      "BalancingController.simulate",
      Types.Role.DESIGNER
    ),
    new Types.RouteDefinition(
      Types.HttpMethod.GET,
      "/decisions",
      "BalancingController.decisions",
      Types.Role.DESIGNER
    ),
    new Types.RouteDefinition(
      Types.HttpMethod.POST,
      "/decisions",
      "BalancingController.createDecisionEntry",
      Types.Role.DESIGNER
    )
  );
}
